irods/exception.py

```diff
@@ -67,8 +69,52 @@ def __init__(self, name, bases, attrs):
             iRODSExceptionMeta.codes[attrs['code']] = self
 
 
+class Errno:
+    def __init__(self,arg0,*args):
+        self.int_code = arg0
+    def __repr__(self):
+        e = self.int_code
+        try:
+            return self.__class__.__name__ + repr(tuple([e, errno.errorcode[e], os.strerror(e)]))
+        except Exception as exc:
+            pass # print(repr(exc))
+        return self.__class__.__name__ + repr(tuple([e,]))
+    def __int__(self):
+        return self.int_code
+       
+
 class iRODSException(six.with_metaclass(iRODSExceptionMeta, Exception)):
-    pass
+
+    def __init__(self,*arg):
+        VE = None
+        if hasattr(self.__class__,'code'):
+           argl = list(arg)
+           if argl and isinstance (argl[-1],Errno):
+               VE = argl.pop()
+        super(iRODSException,self).__init__(*argl)
+        if VE:
+            self.code -= int(VE)
+
+    def __repr__(self,):
+        args = tuple(self.args)
+        if self.code:
+            os_err = abs(self.code) % 1000
+            if os_err: args += (Errno(os_err),)
+        return self.__class__.__name__ + repr(args)
+         
+    ## -- optional - this would let us compare thrown exception with e.g.: 
+    #                UNIX_FILE_WRITE_ERR('some_message',Errno(some_integer))
+    #     
+    def __eq__(self,other):
+        return (
+                isinstance (other,iRODSException) and
+                self._matches_code(other))
+
+    def _matches_code(self, other):
+        code0 = getattr(self,'code',None)
+        code1 = getattr(other,'code',None)
+        return ((code0 is not None) and (code1 is not None) and
+                abs(code0) == abs(code1))
```

Comment:
let's not do this for now - let's just make the repr() work/good.

Reply:
Ok... deleting is definitely easy : )